File: TwoSum/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}

        for i,num in enumerate(nums):
            diff = target - num

            if diff in table.keys(): # found the two sum
                return [i,table[diff]]

            if num not in table.keys(): # store the number in the map
                table[num] = i
        return None

File: TwoSum/TwoSum.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

class Solution {
public:
    std::vector<int> twoSum(std::vector<int>& nums, int target) {

        std::unordered_map<int,int> seen; // Create hashmap

        for (int i = 0; i < nums.size();i++){
            int diff = target - nums[i]; 

            if (seen.find(diff) != seen.end()){ // Check if the difference is in the hashmap / we've already seen this 
                return {i,seen[diff]};
            }

            seen[nums[i]] = i; // Add current number to the hashmap / we've seen the current number
        }
        return {};
    }
};